lib/tasks/scripts/mark_transferred_patients.rb

```diff
@@ -0,0 +1,18 @@
+class MarkTransferredPatient
+  TRANSFER_CANCEL_REASONS = %w[moved_to_private public_hospital_transfer]
+
+  def self.call
+    latest_appointment_per_patient =
+      Appointment
+        .select("DISTINCT ON (patient_id) *")
+        .order("patient_id, updated_at DESC")
+
+    migrated_patient_ids =
+      Appointment
+        .from(latest_appointment_per_patient, "appointments")
+        .where(cancel_reason: TRANSFER_CANCEL_REASONS)
+        .pluck(:patient_id)
+
+    Patient.where(id: migrated_patient_ids).update_all(status: "migrated")
```

Comment:
Would we want to update a patient if they are marked as `dead`?

Also, can you output the number of updated records to the log so that we can monitor how much this updates?

Reply:
Dead patients are already being updated https://github.com/simpledotorg/simple-server/blob/d87efa16e2803e1b63d34de8d7fa8806a59fb214/app/controllers/appointments_controller.rb#L76

Reply:
> Would we want to update a patient if they are marked as `dead`?

Sorry, I meant if a patient's existing status is marked as something non-active (say, dead), would we want to overwrite that w/ `migrated` ?

Reply:
Hmm, that should be unlikely since patients get removed from the overdue list once the appointment status is updated. Both on the app and the dashboard.

Reply:
We can still override the status I think, in case someone does manage to update them again. Shouldn't do any harm.